src/Simplify_Internal.h

```diff
@@ -28,6 +28,20 @@
 namespace Halide {
 namespace Internal {
 
+namespace {
```

Comment:
I think this should not be in an anonymous namespace. This will cause the linker to put multiple copies of this function in the compiled library (normally, functions with inline linkage are not duplicated, except when they are inlined of course).

Reply:
Do you prefer to have the function defined in some translation unit? Because the way the Simplify pass is split I'm not sure where to put it.

Reply:
I think inline is fine (if it had to go somewhere, I guess Simplify.cpp is the right place).